module/users/helper.js

```diff
@@ -1119,6 +1142,150 @@ module.exports = class UsersHelper {
+
+  //             break;
+  //           case "project":
+  //             let projectInfo =
+  //               await improvementProjectService.listProjectOverviewInfo({
+  //                 stats: "false",
+  //                 userToken
+  //               });
+
+  //               resolve({
+  //                 type: type,
+  //                 data: projectInfo.data,
+  //               });
+
+  //             break;
+  //             default:
+  //               throw new Error('Invalid Type passed.')
+  //         }
+  //       } else {
+
+  //         let projectsConsumedByUserStatsAPICall =
+  //           improvementProjectService.listProjectOverviewInfo({
+  //             stats: "true",
+  //             userToken
+  //           });
+
+  //         let observationInfo = surveyService.getObservationInfo(userToken,'true');
+
+  //         Promise.all([
+  //           projectsConsumedByUserStatsAPICall,
+  //           observationInfo,
+  //         ])
+  //           .then((responses) => {
+  //             for(let response of responses){
+  //               if(!response.success)
+  //                 {
+  //                   throw new Error('Failed API calls.')
+  //                 }
+  //             }
+  //             const [response1, response2] =
+  //               responses;
+  //             resolve({
+  //               projectsConsumedByUserStats:response1.data,
+  //               observationWhichUserConsumedStats: response2.data.count,
+  //             });
+  //           })
+  //           .catch((error) => {
+  //             reject(error)
+  //           });
+  //       }
+  //     } catch (error) {
+        
+  //       reject(error);
+  //     }
+  //   });
+  // }
+  static getAllStats({ userId,userToken, type }) {
+    return new Promise(async (resolve, reject) => {
+      try {
+        if (type) {
+          let userOverview;
+          switch (type) {
+            case constants.common.OBSERVATION:
+              userOverview = await surveyService.getObservationInfo(userToken);
+              break;
+            case constants.common.PROJECT:
+              userOverview =
+                await improvementProjectService.listProjectOverviewInfo({
+                  stats: "false",
+                  userToken,
+                });
+              break;
+            case constants.common.SURVEY:
+              userOverview = await surveyService.userSurveyOverView(
+                userToken,
+                false
+              );
+              break;
+            case constants.common.PROGRAM.toLowerCase():
+              userOverview = await programsHelper.userProgram(userId, false);
+              break;
+            default:
+              throw new Error("Invalid Type passed.");
+          }
+          return resolve({
+            success: true,
+            message: `${type} ${constants.apiResponses.FETCH_SUCCESS}`,
+            data: userOverview.data,
+          });
+        } else {
```

Comment:
why else is needed if nothing is passed you can call default in switch case @borkarsaish65

Reply:
done